### ISSUE TITLE
[Camera] Audio is only recorded with first recording on iOS

### ISSUE THREAD

**Issue by hillelcoren** [flutter/flutter#30689]
If the camera plugin is used to capture more than one video only the first video will include audio. The problem can be worked around by calling `camera.initialize();` again however it causes the camera preview to flash and is obviously inefficient. 

Here's a sample app demonstrating the problem. Click record, then stop then play and the audio/video will both work. Click record, stop and then play again and there will only be video.

```Dart
import 'dart:async';
import 'dart:io';
import 'package:path_provider/path_provider.dart';
import 'package:video_player/video_player.dart';
import 'package:flutter/material.dart';
import 'package:camera/camera.dart';

List<CameraDescription> cameras;
Future<void> main() async {
  cameras = await availableCameras();
  runApp(TestApp());
}

class TestApp extends StatefulWidget {
  @override
  TestAppState createState() => TestAppState();
}

class TestAppState extends State<TestApp> {
  VideoPlayerController videoPlayer;
  CameraController camera;
  String path;

  @override
  void initState() {
    super.initState();
    camera = CameraController(
        cameras.firstWhere(
            (camera) => camera.lensDirection == CameraLensDirection.front),
        ResolutionPreset.high)
      ..addListener(() {
        if (mounted) setState(() {});
      })
      ..initialize();
  }

  void record() async {
    final Directory directory = await getApplicationDocumentsDirectory();
    final String folder = '${directory.path}/videos';
    await Directory(folder).create(recursive: true);
    path = '$folder/${DateTime.now().millisecondsSinceEpoch}.mp4';
    await camera.startVideoRecording(path);
  }

  void stopRecording([bool cancel]) async {
    await camera.stopVideoRecording();
    final player = VideoPlayerController.file(File(path));
    await player.initialize();
    setState(() {
      videoPlayer = player;
    });

    // Uncommenting this line 'fixes' the problem
    // await camera.initialize();
  }

  void play() {
    if (!videoPlayer.value.initialized) return;
    videoPlayer
      ..seekTo(Duration())
      ..play();
  }

  void clear() {
    setState(() {
      videoPlayer = null;
    });
  }

  @override
  Widget build(BuildContext context) {
    final value = camera.value;
    if (!value.isInitialized) return SizedBox();
    return MaterialApp(
      home: SafeArea(
        child: Scaffold(
            body: Column(children: [
          Expanded(
              child: AspectRatio(
                  aspectRatio: value.aspectRatio,
                  child: CameraPreview(camera))),
          Material(
            color: Colors.black26,
            elevation: 8,
            child: Row(children: [
              FlatButton(
                child: Text('Record'),
                onPressed: record,
              ),
              FlatButton(
                child: Text('Stop'),
                onPressed: stopRecording,
              ),
              FlatButton(
                child: Text('Play'),
                onPressed: play,
              ),
              FlatButton(
                child: Text('Clear'),
                onPressed: clear,
              ),
            ]),
          ),
          Expanded(
            child: videoPlayer == null
                ? SizedBox()
                : AspectRatio(
                    aspectRatio: videoPlayer.value.aspectRatio,
                    child: VideoPlayer(videoPlayer)),
          )
        ])),
      ),
    );
  }
}
```

**Comment by escamoteur** [comment]
Hi Hillel, sorry for the delay, I just want to check if you could solve this problem.

**Comment by hillelcoren** [comment]
I wasn't able to solve it but found a workaround. I re-initialize the camera after the recording finishes, this makes it ready for use the next time the user tries to record.

**Comment by danielroek** [comment]
According to @TahaTesser in [#70851](https://github.com/flutter/flutter/issues/70851) this issue still persists in iOS 14 with flutter 1.22.4.

**Comment by TahaTesser** [comment]
### Code Sample
https://github.com/flutter/flutter/issues/70851#issue-746805230

```yaml
dependencies:
  flutter:
    sdk: flutter
  camera: ^0.5.8+11
  cupertino_icons: ^0.1.3
  path_provider: ^1.6.24
  uuid: ^2.2.2
  video_player: ^1.0.1
```

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.4, on macOS 11.0.1 20B29 darwin-x64, locale
    en-GB)
    • Flutter version 1.22.4 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 1aafb3a8b9 (7 days ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

 
[✓] Connected device (1 available)            
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios • iOS 14.2

! Doctor found issues in 1 category.
```

</details>


Workaround https://github.com/flutter/flutter/issues/70851#issuecomment-730588921

**Comment by toebgen** [comment]
I ran into this issue as well, the recommended workaround helped! :+1:

In case anybody else comes across this, the "symptom" I discovered this through was slightly different:
A *black screen* in the `VideoPlayerWidget`.

In more detail:
After saving the recording from `camera` into a file, and initialising a `VideoPlayerController` instance with it, the `VideoPlayerWidget` was just showing a *black screen* instead of the recorded video. Interesting enough, when looking a bit more into the controller, it turned out that the duration of the video was several seconds longer than the actual duration of the recording that took place before.
Say, I recorded a 3 sec video, then the duration of the controller was 7 sec. The more often I repeated the recording, the longer this duration became.
It feels like something in the plugin would need some more proper house keeping between recordings.

Maybe this helps others as well, would be great if this was fixed!
